test/others/ns_ext/run.sh

```diff
@@ -0,0 +1,74 @@
+#!/bin/bash
+
+set -x
+
+if [[ "$1" == "pid" ]]; then
+	NS=pid
+else
+	NS=net
+fi
+
+MNT1=test_ns1
+MNT2=test_ns2
+
+trap "cleanup" QUIT TERM INT HUP EXIT
+
+function cleanup()
+{
+	kill -9 $pid $pid2
+	sleep 0.5
+	umount -lf $MNT1 $MNT2 || :
+	rm -f $MNT1 $MNT2
+	rm -f pidfile pidfile2 pidfile3
+}
+
+CRIU=../../../criu/criu
+if [[ "$NS" == "net" ]]; then
+	setsid unshare -n bash -c 'unshare -n sh _run.sh pidfile2 & unshare -n sh _run.sh pidfile3 & ip link add xxx type veth && ip link add mymacvlan1 link xxx type macvlan mode bridge && . _run.sh pidfile' < /dev/zero &> output &
+elif [[ "$NS" == "pid" ]]; then
+	RND1=$RANDOM
+	RND2=$RANDOM
+	setsid unshare -p -f setsid bash -c "setsid sh _run.sh pidfile2 $RND2 & . _run.sh pidfile $RND1" < /dev/zero &> output &
```

Comment:
It would be nice to have a comment which explains these three setsid-s

Reply:
I don't know. Only with a`setsid` at all three places it succeeds with Ubuntu, CentOS and Alpine. I do not understand this part. It seems, especially the alpine busybox behaves differently.